Remove pages ported to dcrdevdocs, add redirects

s-ben opened the pull request:
**NOTE: do not merge until developer documentation is launched.**

In preparation for dev docs launch, removing pages that have been moved to the new dev docs site, adding redirects.

A couple outstanding issues on this PR:

- Removed the #### Testnet reset section from the README, as it was pointing to a page that was moved. Put back in with the link to devdocs page?
- My naive attempt to do nginx redirects to another domain has failed…:/ Before spending more time on it, thought I’d see if anyone knew the solution.

jholdstock commented on conf/nginx.conf:
```diff
@@ -70,4 +70,9 @@ server {
     rewrite ^/proof-of-work/proof-of-work/?$                             $scheme://$http_host/proof-of-work/overview permanent;
     rewrite ^/governance/introduction-to-decred-governance/?$            $scheme://$http_host/governance/overview permanent;
     rewrite ^/contributing/contributing-to-decred/?$                     $scheme://$http_host/contributing/overview permanent;
+    rewrite ^/advanced/address-details/?$                                $https://devdocs.jamieholdstock.com/developer-guides/addresses permanent;
```
URLs need updating now that https://devdocs.decred.org is up

s-ben replied:
Done

jholdstock commented on mkdocs.yml:
```diff
@@ -140,7 +134,6 @@ nav:
   - 'Script Extensions': 'research/script-extensions.md'
   - 'Signature Script Isolation and Fraud Proofs': 'research/signature-script-isolation-and-fraud-proofs.md'
   - 'Block Production Times': 'research/block-production-times.md'
-  - 'Transaction Extensions': 'research/transaction-extensions.md'
```
No redirect for this

s-ben replied:
Added redirect, though not sure if working...

jholdstock commented on mkdocs.yml:
```diff
@@ -113,19 +113,13 @@ nav:
-  - 'Address Details': advanced/address-details.md
-  - 'Using Testnet': 'advanced/using-testnet.md'
   - 'Manual CLI Installation': 'advanced/manual-cli-install.md'
-  - 'Block Header Specifications': 'advanced/block-header-specifications.md'
```
No redirect for this

s-ben replied:
Added redirect, though not sure if working...

jholdstock commented on docs/research/overview.md:
```diff
@@ -16,7 +16,6 @@ Research work in Decred is currently organized around the following subsections:
 * [BLAKE-256 Hash Function](blake-256-hash-function.md)
 * [Script Extensions](script-extensions.md)
 * [Signature Script Isolation and Fraud Proofs](signature-script-isolation-and-fraud-proofs.md)
-* [Transaction Extensions](transaction-extensions.md)
```
Link to devdocs instead of removing link completely?

s-ben replied:
Good idea. Linking to devdocs now

jholdstock commented on docs/faq/wallets-and-seeds.md:
```diff
@@ -64,7 +64,7 @@ dcrctl --wallet getbalance "imported" 0 all
 #### 7. What is the difference between a testnet and mainnet public key address?
 
-A public key address, also called Pay-To-Pubkey (P2Pk), can be identified with its 2-byte prefix which identifies the network and type. A mainnet public key address starts with the letters `Dk` while a testnet public key address starts with the letters `Tk`. For details on address types, see the see the [address details](../advanced/address-details.md) page.  
+A public key address, also called Pay-To-Pubkey (P2Pk), can be identified with its 2-byte prefix which identifies the network and type. A mainnet public key address starts with the letters `Dk` while a testnet public key address starts with the letters `Tk`. For details on address types, see the see the [address details](https://devdocs.decred.org/developer-guides/addresses/) page in the develope ...
```
`see the see the`

s-ben replied:
Fixed

s-ben commented:
@jholdstock have addressed comments. The only thing I'm (potentially) stuck on is the last two redirects added:

rewrite ^/advanced/block-header-specifications/?$                    https://devdocs.decred.org/developer-guides/block-header-specifications     permanent;

rewrite ^/research/transaction-extensions/?$                         https://devdocs.decred.org/developer-guides/transactions/transaction-extensions/     permanent;

These might be working, but locally redirects aren't working for some reason, even though I've managed to get the links matching what's in the redirects...:/ Starting to spin my wheels here, so thought I'd see if you saw the problem right away.

jholdstock commented on docs/research/ticket-selection-algorithm.md:
```diff
@@ -4,5 +4,5 @@ Every block, 5 tickets are pseudorandomly selected from the ticket pool to [vote
 ## Pseudorandom number generation
 
-Tickets are selected from the ticket pool using a deterministic [Pseudorandom Number Generator (PRNG)](https://en.wikipedia.org/wiki/Pseudorandom_number_generator). It is deterministic because the seed input to the PRNG is derived from the hash of the previous block (the block being voted on). Specifically, the seed is a serialization of the [header](../advanced/block-header-specifications.md) of ...
+Tickets are selected from the ticket pool using a deterministic [Pseudorandom Number Generator (PRNG)](https://en.wikipedia.org/wiki/Pseudorandom_number_generator). It is deterministic because the seed input to the PRNG is derived from the hash of the previous block (the block being voted on). Specifically, the seed is a serialization of the [header](/advanced/block-header-specifications/) of the ...
```
```
[header](/advanced/block-header-specifications/)
```

This page is now in devdocs.

s-ben replied:
Fixed

jholdstock commented on docs/wallets/spv.md:
```diff
@@ -18,7 +18,7 @@ Another benefit offered by SPV is an extreme reduction in the time required for
 ## How does SPV work?
 
-At the start of every block added to the Decred blockchain is 180 bytes of data called the [block header](../advanced/block-header-specifications.md). The block header describes key information about the block including the hash of the block, the Merkle root (the sum of all the transaction hashes in the block), and the nonce calculated by the proof-of-work miners. A predetermined filter is also c ...
+At the start of every block added to the Decred blockchain is 180 bytes of data called the [block header](../../advanced/block-header-specifications/). The block header describes key information about the block including the hash of the block, the Merkle root (the sum of all the transaction hashes in the block), and the nonce calculated by the proof-of-work miners. A predetermined filter is also  ...
```
```
[block header](../../advanced/block-header-specifications/)
```

This page is now in devdocs.

s-ben replied:
Fixed

jholdstock commented:
Redirections look fine to me. Here is how I have tested them:

```
# Build docker container
$ bin/build.sh

# Run docker container
$ docker run -d --rm -p 8000:80 decred/dcrdocs:latest

# cURL and output HTTP headers
$ curl -I http://localhost:8000/advanced/block-header-specifications/
```

Here's the output - see the value of the `Location:` header:

```
HTTP/1.1 301 Moved Permanently
Server: nginx
Date: Sat, 02 Nov 2019 10:27:26 GMT
Content-Type: text/html
Content-Length: 162
Connection: keep-alive
Location: https://devdocs.decred.org/developer-guides/block-header-specifications
Expires: Thu, 01 Jan 1970 00:00:01 GMT
Cache-Control: no-cache
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-XSS-Protection: 1; mode=block
Referrer-Policy: no-referrer
```
